# ComTools

## Background

ComTools is a pipeline that focued on analysis about comparative genomics. In this view, it is aimed to deal with data generated by whole genome alignment (by lastz, or lastal and cactus).

##  Main Usage

1. Merging blocks, operating sequecing and filter.

2. Extract or remove assigned features from Maf format file (CDS, gene, mRNA or others).

3. Split alignment in small windows.

4. Construct phylogenetic tree using different model.

5. Estimate convegent sites thourgh alignment results.

## Denpendencies
Environment: Linux, CentOS 6

Perl, version >= 5.32.1

MafFilter, version = 1.3.1 (https://jydu.github.io/maffilter/)

Please make sure they are in $PATH of your computer

##  Preparing Input Files

First we need a MAF format alignment file. you can generate it using lastz, lastal or cactus (cactus recommanded).

For lastal, you could read my previous published pipeline (https://github.com/gotouerina/Comparative-genomics-script/wiki/03.LASTAL-Alignment-Pipeline). 

For lastz, please check the manual.

For cactus, you should follow the instruction of official manual, and extract Maf format file from HAL format file using cactus-hal2maf 

Parameters, for example, "--refGenome $YourRefGenome --chunkSize 500000  --dupeMode single --noAncestors"

## Filter the MAF format file

First we could filter blocks with all species and > 150bp. We provide FilterMaf.pl for this function. Just do

    perl FilterMaf.pl input_MAF output_MAF speciesList 150

you could modifiy 150 to other values to get differet results. 

***Notice that our scripts all consider the first line in speciesList as the reference!***

## Extract CDS from MAF format file

If you have annotation of the reference species, you could extract them using ExtractCDS.pl. Just type

    perl ExtractCDS.pl input_maf output_maf reference_name  gff

## Convert MAF into VCF format

## Calculate Convergent Sites

## Cite
Cite: Please cite website (https://github.com/gotouerina/ComTools), there is no paper published in this stage, but are prepared to publish on Bioaxiv in future.
