# ComTools v0.1 （beta）
![License](https://img.shields.io/badge/license-MIT-yellow) ![Language](https://img.shields.io/badge/language-Perl-brightgreen)![Documentation](https://img.shields.io/badge/documentation-yes-blue)

## Background

ComTools is a pipeline that focues on analysis about comparative genomics. In this view, it is aimed to deal with data generated by whole genome alignment (by lastz, lastal and cactus).

This pipeline is more tolerable than PHAST packages, since there are many format errors when running PHAST for results of cactus.

##  Main Aim

1. Merging blocks, operating sequecing and filter.

2. Extract or remove assigned features(eg. CDS) from Maf format file.

3. Split alignment in small windows.

4. Construct phylogenetic tree.

5. Estimate convegent sites thourgh alignment results.


## Denpendencies
Environment: ***Linux, CentOS 6***

***Perl, version >= 5.32.1***

***MafFilter, version = 1.3.1 (https://jydu.github.io/maffilter/)***

Please make sure they are in $PATH of your computer

## Install

Just type 

        git clone https://github.com/gotouerina/ComTools.git
        

##  Preparing Input Files

First we need a MAF format alignment file. you can generate it using lastz, lastal or cactus (cactus recommanded).

For lastal, you could read my previous published pipeline (https://github.com/gotouerina/Comparative-genomics-script/wiki/03.LASTAL-Alignment-Pipeline). 

For lastz, please check the manual.

For cactus, you should follow the instruction of official manual, and extract Maf format file from HAL format file using cactus-hal2maf 

Parameters, for example, "--refGenome *YourRefGenome* --chunkSize 500000  --dupeMode single --noAncestors"

## Filter the MAF format file
ComTools stored many perl scripts in bin/ folder.

First we could filter blocks with all species and > 150bp. We provide FilterMaf.pl for this function. Just do

        perl FilterMaf.pl input_MAF output_MAF speciesList 150

you could modifiy 150 to other values to get differet results. 

***Notice that our scripts all consider the first line in speciesList as the reference!***

## Extract CDS from MAF format file

If you have annotation of the reference species, you could extract them using ExtractCDS.pl. Just type

        perl ExtractCDS.pl input_maf output_maf reference_name  gff

## Convert MAF into VCF format

        perl GenerateVcf.pl input_maf output_vcf speciesList
        
## Calculate Convergent Sites
Today your should modify the subpoplution in scripts and do

        perl ConvergentSelect.pl input_vcf

## Generate Count Tree
This functoin is used to generate tree for each blocks. And the results could be merged in to species tree by ASTRAL (https://github.com/smirarab/ASTRAL).

Generally Type:

        perl GenerateCountTree.pl input_maf OurgroupName output_tree

The output_tree contains x rows (x is the blocks of input MAF file). 

Then Type : 

        java -jar /path/to/astral.5.7.8.jar -i output_tree

You can get the final species tree.

## Convert MAF into MSMC format
This function is under development.
## Convert MAF into PLINK format
This function is under development.

## Scripts
Scripts that will help in analysis were included in "utils" folder. 

split_lst.pl was prepared to split "lst" format file into each chr.

convert.pl was prepared to convert name in one gff3 according to a "convert.txt", the format of it can be seen in example folder.


## Cite
Cite: Please cite website (https://github.com/gotouerina/ComTools) when you use this pipeline, there is no paper published now.

## Affiliations
State Key Laboratory of Herbage Improvement and Grassland Agro-ecosystems, College of Ecology, Lanzhou University, Lanzhou 730000, China.

## Getting Help
You could write your problems in github issue page, or you could e-mail me at kogoorimasaki@outlook.com (**Recommanded**).
